app/main.py

```diff
@@ -1,8 +1,39 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self, comfort_class: int, clean_mark: int, brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self, distance_from_city_center: float, clean_power: int,
+                 average_rating: float, count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def rate_service(self, new_rating: float) -> None:
+        total_rating = self.average_rating * self.count_of_ratings
+        total_rating += new_rating
+        self.count_of_ratings += 1
+        self.average_rating = round(total_rating / self.count_of_ratings, 1)
+
+    def wash_single_car(self, car: Car) -> None:
+        if car.clean_mark < self.clean_power:
+            car.clean_mark = self.clean_power
+
+    def calculate_washing_price(self, car: Car) -> float:
+        if car.clean_mark < self.clean_power:
+            cost = round(car.comfort_class * (self.clean_power - car.clean_mark)
+                         * (self.average_rating / self.distance_from_city_center), 1)
+            return cost
+        return 0.0
+
+    def serve_cars(self, cars: list) -> float:
+        income = 0.0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                self.wash_single_car(car)
+                income += self.calculate_washing_price(car)
```

Comment:
In the `serve_cars` method, the `calculate_washing_price` method should be called before `wash_single_car`. This is because washing the car changes its `clean_mark`, which affects the price calculation. Ensure the price is calculated based on the car's condition before washing.